modules/tinymce/src/core/test/ts/webdriver/delete/SelectionEnabledDeleteContentTest.ts

```diff
@@ -0,0 +1,386 @@
+      TinyAssertions.assertContent(editor, '<p><iframe src="https://www.youtube.com/embed/8aGhZQkoFbQ" width="560" height="314" allowfullscreen="allowfullscreen"></iframe></p>');
+
+      setMode(editor, 'design');
+      TinySelections.select(editor, 'span.mce-preview-object', []);
+      assertFakeSelection(editor, true);
+      await RealKeys.pSendKeysOn('iframe => body', [ RealKeys.backspace() ]);
+      TinyAssertions.assertContent(editor, '');
+    });
+
+    it('TINY-10891: Deleting mediaembed element should not be permitted', async () => {
+      const editor = hook.editor();
+      editor.setContent(iframeMediaEmbedHtml);
+
+      setMode(editor, 'testmode');
+      TinySelections.select(editor, 'div[data-ephox-embed-iri]', []);
+      assertFakeSelection(editor, true);
+      await RealKeys.pSendKeysOn('iframe => body', [ RealKeys.backspace() ]);
+      TinyAssertions.assertContent(editor, iframeMediaEmbedHtml);
+
+      setMode(editor, 'design');
+      TinySelections.select(editor, 'div[data-ephox-embed-iri]', []);
+      assertFakeSelection(editor, true);
+      await RealKeys.pSendKeysOn('iframe => body', [ RealKeys.backspace() ]);
+      TinyAssertions.assertContent(editor, '');
+    });
+
+    it('TINY-10891: Deleting pre block should not be permitted', async () => {
+      const editor = hook.editor();
+      editor.setContent(preCodeSampleHtml);
+
+      setMode(editor, 'testmode');
+      TinySelections.select(editor, 'pre', []);
+      assertFakeSelection(editor, true);
+      await RealKeys.pSendKeysOn('iframe => body', [ RealKeys.backspace() ]);
+      TinyAssertions.assertContent(editor, preCodeSampleHtml);
+
+      setMode(editor, 'design');
+      TinySelections.select(editor, 'pre', []);
+      assertFakeSelection(editor, true);
+      await RealKeys.pSendKeysOn('iframe => body', [ RealKeys.backspace() ]);
+      TinyAssertions.assertContent(editor, '');
+    });
+
+  });
+
+  it('TINY-10891: Deleting div element (tableofcontents) block should not be permitted', async () => {
+    const editor = hook.editor();
+    editor.setContent(tableOfContentHtml);
+
+    setMode(editor, 'testmode');
+    TinySelections.select(editor, 'div', []);
+    assertFakeSelection(editor, true);
+    await RealKeys.pSendKeysOn('iframe => body', [ RealKeys.backspace() ]);
+    TinyAssertions.assertContent(editor, [ '<div class="mce-toc" contenteditable="false">',
+      '<h2 contenteditable="true">Table of Contents</h2>',
+      '<ul>',
+      '<li><a href="#mcetoc_">Heading</a></li>',
+      '</ul>',
+      '</div>' ].join('\n'));
+
+    setMode(editor, 'design');
+    TinySelections.select(editor, 'div', []);
+    assertFakeSelection(editor, true);
+    await RealKeys.pSendKeysOn('iframe => body', [ RealKeys.backspace() ]);
+    TinyAssertions.assertContent(editor, '');
+  });
+
+  it('TINY-10891: Deleting table content should not be permitted in cusorEnabled mode', async () => {
```

Comment:
There appears to be a few other test cases with the old cursorEnabled name
```suggestion
  it('TINY-10891: Deleting table content should not be permitted in selectionEnabled mode', async () => {
```